Before: 92a69caee8b19d2a677fc4a5478325e1219a6de3308c321abb64862dbc0f128f
After: 29e8c6dc90c6518056afaef3803b756d5d1872288a8a6546befb35ffda476a46
Patch: c68578a876cbf3beb069626469c0f08e22091b7b0e2ce0829f7320ddfe3dc7e7
--- %project_name%.gemspec.tt	2012-09-04 15:04:03.414401052 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,21 +0,0 @@
-# encoding: utf-8
-require File.expand_path('../lib/<%= project_name %>/version', __FILE__)
-
-Gem::Specification.new do |gem|
-  gem.name        = "<%= project_name %>"
-  gem.version     = <%= project_module %>::VERSION.dup
-  gem.date        = Date.today.to_s
-  #gem.summary     = %q{TODO: Write a gem summary}
-  #gem.description = %q{TODO: Write a gem description}
-  <%= author_name_comment  %>gem.authors     = ["<%= author_name  || 'TODO' %>"]
-  <%= author_email_comment %>gem.email       = ["<%= author_email || 'TODO' %>"]
-  #gem.homepage    = "https://github.com/username/<%= project_name %>"
-
-  gem.add_runtime_dependency 'corvid'
-
-  gem.files         = File.exists?('.git') ? `git ls-files`.split($\) : \
-                      Dir['**/*'].reject{|f| !File.file? f or %r!^(?:target|resources/latest)/! === f}.sort
-  gem.require_paths = %w[lib]
-  gem.test_files    = gem.files.grep(/^test\//)
-end
-
--- Gemfile	2012-09-04 11:37:06.610409762 +1000
+++ Gemfile	2012-09-04 15:45:19.574358890 +1000
@@ -1,5 +1,4 @@
 source :rubygems
-gemspec
 
 # Load Corvid dependencies
 eval File.read(File.expand_path '../.corvid/Gemfile', __FILE__)
--- bin/%plugin_name%.tt	2012-09-04 15:40:24.926105873 +1000
+++ bin/%plugin_name%.tt	2012-09-04 15:45:19.574358890 +1000
@@ -17,5 +17,5 @@
 require 'corvid/cli/plugin'
 Corvid::CLI::Plugin.start project_root:        project_root,
                           plugin_require_path: '<%= require_path %>',
-                          plugin_class_name:   '<%= full_class_name %>'
+                          plugin_class_name:   '<%= class_name %>'
 
--- corvid-features/corvid.rb	2012-09-04 15:30:07.712796459 +1000
+++ corvid-features/corvid.rb	2012-09-04 15:45:19.574358890 +1000
@@ -13,18 +13,5 @@
   copy_executable 'bin/yard'
   copy_executable 'bin/yardoc'
   copy_executable 'bin/yri'
-  update 7
+  empty_directory 'lib'
 }
-
-update {|ver|
-  case ver
-  when 7
-    template2       'lib/%project_name%/version.rb.tt'
-    template2       '%project_name%.gemspec.tt'
-  end
-}
-
-# Template vars
-def author_name_comment;  author_name  ? '' : '#' end
-def author_email_comment; author_email ? '' : '#' end
-
--- lib/%project_name%/%feature_name%_feature.rb.tt	2012-09-04 15:45:16.304341263 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,12 +0,0 @@
-require 'corvid/feature'
-
-module <%= project_module %>
-  class <%= class_name %> < Corvid::Feature
-
-    since_ver <%= since_ver %>
-
-    rake_tasks {
-    }
-
-  end
-end
--- lib/%project_name%/%plugin_name%_plugin.rb.tt	2012-09-04 15:40:02.825987052 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,37 +0,0 @@
-require 'corvid/plugin'
-#require 'corvid/extension'
-
-module <%= project_module %>
-  class <%= class_name %> < Corvid::Plugin
-    # Plugins can be extensions but are not by default.
-    #include Corvid::Extension
-
-    # The name of the plugin. Must conform to format enforced by {Corvid::NamingPolicy#validate_plugin_name!}.
-    #
-    # @return [String] The plugin name.
-    name '<%= plugin_name %>'
-
-    # The path for Ruby to require in order to load this plugin.
-    #
-    # @return [String] The path to require, usually relative to your lib dir.
-    require_path '<%= require_path %>'
-
-    # The path to the directory containing the plugin's resources.
-    #
-    # @return [String] An absolute path.
-    resources_path File.expand_path('../../../resources', __FILE__)
-
-    # A manifest of all features provided by the plugin.
-    #
-    # @return [Hash<String,Array<String>>] A hash with keys being feature names, and the values being a 2-element
-    #   array of the feature's require-path, and class name, respectively.
-    feature_manifest ({
-    })
-
-    # A list of features to install automatically when the plugin itself is installed.
-    #
-    # @return [Array<String>] An array of feature names. Do not include the plugin prefix.
-    auto_install_features %w[]
-
-  end
-end
--- lib/%project_name%/version.rb.tt	2012-09-04 10:40:11.080074440 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,5 +0,0 @@
-module <%= project_module %>
-  # The current version of <%= project_module %>.
-  VERSION = "0.0.1".freeze
-end
-
--- /dev/null	2012-09-03 14:23:42.005077313 +1000
+++ lib/corvid/%name%_feature.rb.tt	2012-09-04 15:45:19.561025484 +1000
@@ -0,0 +1,10 @@
+require 'corvid/feature'
+
+class <%= class_name %> < Corvid::Feature
+
+  since_ver <%= since_ver %>
+
+  rake_tasks {
+  }
+
+end
--- /dev/null	2012-09-03 14:23:42.005077313 +1000
+++ lib/corvid/%name%_plugin.rb.tt	2012-09-04 15:45:19.574358890 +1000
@@ -0,0 +1,35 @@
+require 'corvid/plugin'
+#require 'corvid/extension'
+
+class <%= class_name %> < Corvid::Plugin
+  # Plugins can be extensions but are not by default.
+  #include Corvid::Extension
+
+  # The name of the plugin. Must conform to format enforced by {Corvid::NamingPolicy#validate_plugin_name!}.
+  #
+  # @return [String] The plugin name.
+  name '<%= name %>'
+
+  # The path for Ruby to require in order to load this plugin.
+  #
+  # @return [String] The path to require, usually relative to your lib dir.
+  require_path '<%= require_path %>'
+
+  # The path to the directory containing the plugin's resources.
+  #
+  # @return [String] An absolute path.
+  resources_path File.expand_path('../../../resources', __FILE__)
+
+  # A manifest of all features provided by the plugin.
+  #
+  # @return [Hash<String,Array<String>>] A hash with keys being feature names, and the values being a 2-element
+  #   array of the feature's require-path, and class name, respectively.
+  feature_manifest ({
+  })
+
+  # A list of features to install automatically when the plugin itself is installed.
+  #
+  # @return [Array<String>] An array of feature names. Do not include the plugin prefix.
+  auto_install_features %w[]
+
+end
--- resources/latest/corvid-features/%feature_name%.rb.tt	2012-09-04 10:18:09.853186514 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,20 +0,0 @@
-# Requirements that must be satisfied before the feature can be installed.
-# @return [nil, String, Hash<String,Fixnum|Range|Array<Fixnum>>, Array] Requirements that can be provided to
-#   {RequirementValidator}.
-# @see RequirementValidator#add
-#
-#requirements
-
-install {
-  # empty_directory 'lib'
-  # copy_file       'hello.txt'
-  # copy_executable 'bin/hello'
-}
-
-# update {|ver|
-#   case ver
-#   when 2
-#     update stuff here
-#     empty_directory 'lib'
-#   end
-# }
--- /dev/null	2012-09-03 14:23:42.005077313 +1000
+++ resources/latest/corvid-features/%name%.rb.tt	2012-09-04 15:45:19.561025484 +1000
@@ -0,0 +1,20 @@
+# Requirements that must be satisfied before the feature can be installed.
+# @return [nil, String, Hash<String,Fixnum|Range|Array<Fixnum>>, Array] Requirements that can be provided to
+#   {RequirementValidator}.
+# @see RequirementValidator#add
+#
+#requirements
+
+install {
+  # empty_directory 'lib'
+  # copy_file       'hello.txt'
+  # copy_executable 'bin/hello'
+}
+
+# update {|ver|
+#   case ver
+#   when 2
+#     update stuff here
+#     empty_directory 'lib'
+#   end
+# }
--- /dev/null	2012-09-03 14:23:42.005077313 +1000
+++ test/spec/%name%_plugin_spec.rb.tt	2012-09-04 15:45:19.574358890 +1000
@@ -0,0 +1,12 @@
+# encoding: utf-8
+require_relative '../bootstrap/spec'
+require 'corvid/test/resource_patch_tests'
+require '<%= require_path %>'
+
+describe <%= class_name %> do
+  include Corvid::ResourcePatchTests
+
+  include_resource_patch_tests
+
+  include_feature_update_install_tests
+end
--- test/spec/%plugin_name%_plugin_spec.rb.tt	2012-09-04 15:40:16.179392177 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,12 +0,0 @@
-# encoding: utf-8
-require_relative '../bootstrap/spec'
-require 'corvid/test/resource_patch_tests'
-require '<%= require_path %>'
-
-describe <%= full_class_name %> do
-  include Corvid::ResourcePatchTests
-
-  include_resource_patch_tests
-
-  include_feature_update_install_tests
-end
